--- origsrc/libXaw-1.0.11/configure.ac	2012-06-01 22:31:36.000000000 -0500
+++ src/libXaw-1.0.11/configure.ac	2012-10-15 22:36:42.436636800 -0500
@@ -98,6 +98,15 @@ AC_CHECK_HEADERS([wctype.h wchar.h widec
 # Checks for functions
 AC_CHECK_FUNCS([iswalnum getpagesize])
 
+# Link with winsock if mingw target
+case $host_os in
+	*mingw*)
+		AC_HAVE_LIBRARY(ws2_32)
+	;;
+	*)
+	;;
+esac
+
 AC_CONFIG_FILES([Makefile
 		include/Makefile
 		man/Makefile
--- origsrc/libXaw-1.0.11/src/TextAction.c	2012-06-01 22:31:37.000000000 -0500
+++ src/libXaw-1.0.11/src/TextAction.c	2012-10-15 22:34:08.036423800 -0500
@@ -44,6 +44,9 @@ in this Software without prior written a
 #include <X11/Xaw/TextP.h>
 #include <X11/Xaw/TextSrcP.h>
 #include <X11/Xaw/XawImP.h>
+#ifdef _WIN32
+#include <X11/Xwinsock.h>
+#endif
 #include "Private.h"
 #include "XawI18n.h"
 
